src/TargetingSystem.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "GameObject.h"
+
+
+#include <string>
+#include <map>
+#include <vector>
+#include <fstream>
+#include <cstdlib>
+#include <ctime>
```

Comment:
I feel like some of these can be moved to the cpp file. It's better to have includes in the cpp files to speed up compile times.